submissions/5mountains/dom-api-task/js/main.js

```diff
@@ -0,0 +1,102 @@
+const styles = {
+    menuLink: 'menu__link',
+    menuItem: 'menu__item',
+    heroItem: 'hero__item',
+    heroImg: 'hero__image',
+    itemIcon: 'item__icon',
+    itemTitle: 'item__caption',
+    itemPrice: 'item__price',
+    itemInfo: 'item__description'
+};
+
+async function fetchAsync () {
+    let response = await fetch('./data/data.json');
+    let data = await response.json();
+    return data;
+}   
+  
+fetchAsync()
+    .then(data => {
+        let details = createHtml(data);
+        drawPage(details);
+        toggleTabs();
+        animation();
+    })
+    .catch(reason => console.log(reason.message));
+
+function createHtml(data) {
+    const menuItems = document.createDocumentFragment();
+    const heroItems = document.createDocumentFragment();
+    data.pancakes.forEach((elem, index) => {
+        const navContent = generateNavContent(elem, index); 
+        const heroContent = generateHeroContent(elem, index); 
+        menuItems.appendChild(navContent);
+        heroItems.appendChild(heroContent);
+    });
+    return {
+        menuItems,
+        heroItems
+    };
+}
+
+function generateNavContent(elem, index) {
+    let navItem = generateElement('li', styles.menuItem);
+    let navLink = `<a class=${styles.menuLink} href='#' id=link${index+1}>${elem.title}</a>`;
+    navItem.insertAdjacentHTML('afterbegin', navLink);
+    return navItem;
+}
+
+function generateHeroContent(elem, index) {
+    let sectionItem = generateElement('div', styles.heroItem);
+    sectionItem.setAttribute('id', `tab${index+1}`);
+    let sectionItemContent = `
+        <div class=${styles.itemPrice}>Price for portion ${elem.price} $</div>
+        <figure class=${styles.heroImg} id='tab${index+1}'>
+            <img class=${styles.itemIcon} src=${elem.path} />
+            <figcaption class=${styles.itemTitle}>Pancakes ${elem.title}</figcaption>
+        </figure>
+        <div class=${styles.itemInfo}>${elem.description}</div>
+    `;
+    sectionItem.insertAdjacentHTML('afterbegin', sectionItemContent);
+    return sectionItem;
+}
+
+function generateElement(elementName, className) {
+    let item = document.createElement(elementName);
+    item.classList.add(className);
+    return item;
+}
+
+function drawPage(details) {
+    document.querySelector('.nav__menu').appendChild(details.menuItems);
+    document.querySelector('.hero').appendChild(details.heroItems);
+}
+
+function toggleTabs(){
+    document.querySelectorAll('.menu__link').forEach(elem => {
```

Comment:
Use one event listener on the parent element, please. More info [here](https://learn.javascript.ru/event-delegation)

Reply:
ready